src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -593,7 +594,7 @@ export interface ITerminalInstance {
 	/**
 	 * The xterm.js instance for this terminal.
 	 */
-	readonly xterm?: IXtermTerminal;
+	readonly xterm?: XtermTerminal;
```

Comment:
this is necessary to get the buffer content, terminal element, and more

Reply:
I think `ITerminalContribution` is what you're after, that exposes more of the terminal internals that we don't want visible to everything that can access `ITerminalInstance`

Reply:
we now create the widget in terminal instance